db/compaction/compaction_job.cc

```diff
@@ -943,6 +956,15 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
     assert(sub_compact->builder != nullptr);
     assert(sub_compact->current_output() != nullptr);
     sub_compact->builder->Add(key, value);
+    if (paranoid_file_checks_) {
+      auto curr = sub_compact->current_output();
+      assert(curr != nullptr);
+      // Generate a rolling 64-bit hash of the key and values
+      curr->paranoid_hash = Hash64(key.data(), key.size(), curr->paranoid_hash);
+      curr->paranoid_hash =
+          Hash64(value.data(), value.size(), curr->paranoid_hash);
+    }
```

Comment:
Can we make this logic (Add() and paranoid check) into sub_compact->builder, so that in case in the future we add K/Vs in another place, we won't miss this paranoid check.

Btw, I remember there is another place we call sub_compact->builder->Add(), do we need to cover that path too?

Reply:
Are you suggesting TableBuilder::Add do this calculation?  That could be done, but then all of the implementations of TableBuilder would need to perform the calculation (or pass it to a higher layer).  

There are other places Builder::Add is called, to add things like range deletes.  Should those also be covered by this case?

Reply:
Sorry I meant to move to sub_compact. I didn't mean to move into Builder::Add(). I hope this hash check is independent of table builder implementation.